File: gadapter/src/main/java/com/zeyad/gadapter/GenericViewHolder.java
package com.zeyad.gadapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public abstract class GenericViewHolder extends RecyclerView.ViewHolder {

    public GenericViewHolder(View itemView) {
        super(itemView);
    }

    public abstract void bindData(@NonNull Object data, boolean itemSelected, int position, boolean isEnabled);
}

File: gadapter/src/main/java/com/zeyad/gadapter/GenericViewHolder.kt
package com.zeyad.gadapter

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.extensions.LayoutContainer

abstract class GenericViewHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView), LayoutContainer {

    override val containerView: View?
        get() = itemView

    abstract fun bindData(data: T, position: Int, isItemSelected: Boolean, isEnabled: Boolean)
}
